fili-core/src/main/java/com/yahoo/bard/webservice/metadata/SegmentMetadataLoader.java

```diff
@@ -156,7 +156,6 @@ public void invoke(JsonNode rootNode) {
                 if (segmentMetadata.isEmpty()) {
                     LOG.warn("Empty segment metadata detected when loading table '{}'", table.getFactTableName());
                 }
-                table.resetColumns(segmentMetadata, dimensionDictionary);
```

Comment:
Well, seeing as how removing this line means that this class _will not work_, we have two choices:

1. Remove the class completely (so people relying on it get compile time rather than runtime errors).

2. Find some way of getting the segments pulled out by `SegmentMetadataLoader` and shoved into the tables in the new scheme to preserve backwards compatibility.

I haven't dug deeply enough into the code yet to see how feasible number 2 is. Number 1 will make for some migration pain, but I think migration pain is unavoidable at this point. The only question is whether that pain should be up-front and found during migration, or subtle and found during production.